appium/lib/screen-object.rb

```diff
@@ -69,81 +66,80 @@ def enter
     #pending implementation
   end
 
-  def scroll_down_find(locator,locator_value,num_loop = 15)
-    scr = driver.window_size
-    screenHeightStart = (scr.height) * 0.5
-    scrollStart = screenHeightStart.to_i
-    screenHeightEnd = (scr.height) * 0.2
-    scrollEnd = screenHeightEnd.to_i
-    for i in 0..num_loop
-      begin
-        if (driver.find_element(locator,locator_value).displayed?)
-          break
-        end
-      rescue
-        driver.swipe(:start_x => 0,:start_y => scrollStart,:end_x =>0,:end_y =>scrollEnd,:touchCount => 2,:duration => 0)
-        false
-      end
-    end
+  def gesture(x1,y1,x2,y2,duration)
+    Appium::TouchAction.new($driver).swipe(start_x: x1, start_y: y1, end_x: x2, end_y: y2,duration: duration).perform
+  rescue
+    raise("Error during gesture")
   end
 
-  def scroll_down_click(locator,locator_value,num_loop = 15)
-    scr = driver.window_size
-    screenHeightStart = (scr.height) * 0.5
-    scrollStart = screenHeightStart.to_i
-    screenHeightEnd = (scr.height) * 0.2
-    scrollEnd = screenHeightEnd.to_i
-    for i in 0..num_loop
-      begin
-        if (driver.find_element(locator,locator_value).displayed?)
-          driver.find_element(locator,locator_value).click
-          break
-        end
-      rescue
-        driver.swipe(:start_x => 0,:start_y => scrollStart,:end_x =>0,:end_y =>scrollEnd,:touchCount => 2,:duration => 0)
-        false
-      end
+  def scroll(direction = :down, duration = 1000)
+    size = driver.window_size
+    x = size.width/2
+    y = size.height/2
+    case direction
+    when :up then loc = [x, y*0.5, x, (y + (y*0.5)),duration]
+    when :down then loc =[x, y, x, y * 0.5,duration]
+    when :left then loc = [x * 0.6, y, x * 0.3, y,duration]
+    when :right then loc =[x * 0.3, y, x * 0.6, y,duration]
+    else
+      raise('Only upwards and downwards scrolling are supported')
```

Comment:
This message is untrue, make the message clear that `scroll` only supports `:up, :down, :left, :right`